File: ecommerce-module/src/main/java/com/sdl/webapp/ecommerce/ProductCatalog.java
package com.sdl.webapp.ecommerce;

import com.sdl.webapp.common.api.localization.Localization;
import com.sdl.webapp.ecommerce.model.Category;
import com.sdl.webapp.ecommerce.model.Product;
import com.sdl.webapp.ecommerce.model.entity.ProductContent;

import java.util.List;

/**
 * Product Catalog
 *
 * @author nic
 */
public interface ProductCatalog {

    // TODO: Decouple this further and have abstract Product & Category Reference interfaces?? I.e. instead of having string references???
    // ProductRef -> which can be a string or something else...
    // Which can created through an ECL reference maybe????


    Category getCategory(String id) throws ECommerceException;

    Product getProduct(String id) throws ECommerceException;

    List<ProductContent> getProductContent(Category category, Localization localization) throws ECommerceException;

}

File: ecl/ecommerce-ecl-framework/ecommerce-ecl-framework/ProductCatalog.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SDL.ECommerce.Ecl
{
    public interface ProductCatalog
    {
        Category GetAllCategories();

        IList<string> GetProductIds(string categoryId);        

        Product GetProduct(string id);
    }
}
